cli/cmd/generate_gke.go

```diff
@@ -0,0 +1,486 @@
+		"specify the organization id (only set if organization_integration is set)")
+	generateGkeTfCommand.PersistentFlags().BoolVar(
+		&GenerateGkeCommandState.OrganizationIntegration,
+		"organization_integration",
+		false,
+		"enable organization integration")
+	generateGkeTfCommand.PersistentFlags().StringVar(
+		&GenerateGkeCommandState.Prefix,
+		"prefix",
+		"",
+		"prefix that will be used at the beginning of every generated resource")
+	generateGkeTfCommand.PersistentFlags().StringVar(
+		&GenerateGkeCommandState.ProjectId,
+		"project_id",
+		"",
+		"specify the project id to be used to provision lacework resources (required)")
+	generateGkeTfCommand.PersistentFlags().StringVar(
+		&GenerateGkeCommandState.ServiceAccountCredentials,
+		"service_account_credentials",
+		"",
+		"specify service account credentials JSON file path (leave blank to make use of google credential ENV vars)")
+	generateGkeTfCommand.PersistentFlags().StringVar(
+		&GenerateGkeCommandState.WaitTime,
+		"wait_time",
+		"",
+		"amount of time to wait before the next resource is provisioned")
+	generateGkeTfCommand.PersistentFlags().BoolVar(
+		&GenerateGkeCommandExtraState.TerraformApply,
+		"apply",
+		false,
+		"run terraform apply without executing plan or prompting",
+	)
+	generateGkeTfCommand.PersistentFlags().StringVar(
+		&GenerateGkeCommandExtraState.Output,
+		"output",
+		"",
+		"location to write generated content (default is ~/lacework/gcp)",
+	)
+}
+
+func gkeConfigIsEmpty(g *gcp.GenerateGkeTfConfigurationArgs) bool {
+	return g.ServiceAccountCredentials == "" &&
+		g.OrganizationId == "" &&
+		g.LaceworkProfile == ""
+}
+
+func writeGkeGenerationArgsCache(a *gcp.GenerateGkeTfConfigurationArgs) {
+	if !gkeConfigIsEmpty(a) {
+		if a.ExistingServiceAccount.IsPartial() {
+			a.ExistingServiceAccount = nil
+		}
+		cli.WriteAssetToCache(CachedGcpAssetIacParams, time.Now().Add(time.Hour*1), a)
+	}
+}
+
+func promptGkeGenerate(config *gcp.GenerateGkeTfConfigurationArgs, existingServiceAccount *lwgenerate.ServiceAccount, extraState *GkeGenerateCommandExtraState) error {
+	if cli.InteractiveMode() {
+		defer writeGkeGenerationArgsCache(config)
+		defer extraState.writeCache()
+	}
+
+	if existingServiceAccount.Name != "" ||
+		existingServiceAccount.PrivateKey != "" {
+		config.ExistingServiceAccount = existingServiceAccount
+	}
+
+	if err := SurveyMultipleQuestionWithValidation(
+		[]SurveyQuestionWithValidationArgs{
+			{
+				Prompt:   &survey.Input{Message: QuestionGkeProjectID, Default: config.ProjectId},
+				Required: true,
+				Response: &config.ProjectId,
+			},
+			{
+				Prompt:   &survey.Confirm{Message: QuestionGkeOrganizationIntegration, Default: config.OrganizationIntegration},
+				Response: &config.OrganizationIntegration,
+			},
+			{
+				Prompt:   &survey.Input{Message: QuestionGkeOrganizationID, Default: config.OrganizationId},
+				Checks:   []*bool{&config.OrganizationIntegration},
+				Required: true,
+				Response: &config.OrganizationId,
```

Comment:
Maybe a validation here too? We haven't done it but I wonder if we should add one 🤔

Reply:
I wasn't able to find anything around orgID validation.  Could we have this as a new Jira task?